backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/pporder/process/WEBUI_PP_Order_PrintLabel.java

```diff
@@ -73,55 +79,73 @@ protected ProcessPreconditionsResolution checkPreconditionsApplicable()
 	protected String doIt()
 	{
 
+		// create selection
+		final ImmutableSet<HuId> distinctHuIds = retrieveSelectedHuIds();
+		DB.createT_Selection(getPinstanceId(), distinctHuIds, ITrx.TRXNAME_None);
+
 		// print
-		final PPOrderLineRow row = getSingleSelectedRow();
-		final ReportResult label = printLabel(row);
+		final ReportResult label = printLabel();
 
 		// preview
-		getResult().setReportData(
-				label.getReportContent(),
-				buildFilename(row),
-				OutputType.PDF.getContentType());
+		getResult().setReportData(label.getReportContent(), buildFilename(), OutputType.PDF.getContentType());
 
 		return MSG_OK;
 
 	}
 
-	private ReportResult printLabel(@NonNull final PPOrderLineRow row)
+	private ImmutableSet<HuId> retrieveSelectedHuIds()
 	{
-		final PInstanceRequest pinstanceRequest = createPInstanceRequest(row);
+		final DocumentIdsSelection selectedRowIds = getSelectedRowIds();
+
+		final ImmutableList<PPOrderLineRow> selectedRows = getView().streamByIds(selectedRowIds)
+				.collect(ImmutableList.toImmutableList());
+
+		final List<HuId> huIds = new ArrayList<HuId>();
+
+		selectedRows.forEach(row -> {
```

Comment:
for sake of readability pls use a classic `for`.

about: `PPOrderLineRow::getHuId`
* check and consider only those lines which are about HUs, i.e. ppOrderLineRow.getType().isHUOrHUStorage()
* guard against null for getHuId()

* add the huIds directly to a Set, instead of adding them to a list and then converting to a Set.